packages/tables/src/Concerns/HasColumns.php

```diff
@@ -134,27 +129,14 @@ public function setColumnValue(string $column, string $record, $input): ?array
             ];
         }
 
-        if ($columnRelationship = $column->getRelationship($record)) {
-            $record = $columnRelationship->getResults();
-            $columnName = $column->getRelationshipTitleColumnName();
-        } elseif (
-            $this instanceof HasRelationshipTable &&
-            (($tableRelationship = $this->getRelationship()) instanceof BelongsToMany) &&
-            in_array($columnName, $tableRelationship->getPivotColumns())
-        ) {
-            $record = $record->{$tableRelationship->getPivotAccessor()};
-        } else {
-            $columnName = (string) Str::of($columnName)->replace('.', '->');
-        }
-
-        if (! ($record instanceof Model)) {
-            return null;
-        }
-
-        $record->setAttribute($columnName, $input);
-        $record->save();
+        $state = $column->evaluate($column->getSaveStateUsing(), [
+            'state' => $input,
+            'table' => $this->getCachedTable(),
+        ]);
 
-        return null;
+        return [
+            'state' => $state,
```

Comment:
Whats the reason to always return within this `state` key from this method?

Reply:
Cant we return the state without the array like previously?

Reply:
We didn't return the sate at all previously. No sure why I used an array. Probably had something else in there and removed it.